js/pokemon.js

```diff
@@ -1146,3 +1208,30 @@ function loadState() {
         }
     }
 }
+
+Array.prototype.equals = function (array) {
+    // if the other array is a falsy value, return
+    if (!array) {
+        return false;
+    }
+    
+    if (this.length > array.length) {
```

Comment:
You should immediately return false in this scenario. If the arrays aren't the same length then they're not equal